tools/cli-core/src/util.ts

```diff
@@ -170,3 +177,44 @@ export const generateJSONFile = ( filePath: string, data: unknown ) => {
 	);
 	return writeFile( filePath, json );
 };
+
+/**
+ * Get plugin data from main PHP file in JSON format.
+ *
+ * @param {string} tmpRepoPath    - Path to repo.
+ * @param {string} pathToMainFile - Path to plugin's main PHP file.
+ * @param {string} hashOrBranch   - Hash or branch to checkout.
+ * @return {Object} - plugin data in JSON format.
+ */
+export const getPluginData = async (
+	tmpRepoPath: string,
+	pathToMainFile: string,
+	hashOrBranch: string
+) => {
+	const git = simpleGit( { baseDir: tmpRepoPath } );
+	await git.checkout( [ hashOrBranch ] );
+
+	const mainFile = join( tmpRepoPath, pathToMainFile );
+
+	const content = await readFile( mainFile, 'utf-8' );
+	const regex = /<\?php\n\/\*\*\n([\S\s]+?) \*\//;
+	const match = content.match( regex );
+
+	if ( ! match || match.length < 2 ) {
+		Logger.error( 'Failed to find plugin data!' );
```

Comment:
As a pattern in helpers and shared utils, don't use the Logger directly. Throw an error instead. Then catch in the workflow and use the Logger there. It just gives you much better control over handling error states from workflows.

Consider a situation where you wanted to try getPluginData, but maybe if it failed you didn't want to crash but do something else instead, in that case you would put a try/catch around getPluginData and handle it some other way

Reply:
Good call @samueljseay, done in https://github.com/woocommerce/woocommerce/pull/34951/commits/55f8515ddcdd590ef132908e82436b4226c1734f